modules/contact-form/js/gutenblock.jsx

```diff
@@ -0,0 +1,568 @@
+/*global jQuery _ wp*/
+/** @jsx wp.element.createElement */
+/** @format */
+
+/**
+ * External dependencies
+ */
+import {
+	PanelBody,
+	TextControl,
+	TextareaControl,
+	CheckboxControl,
+	ToggleControl,
+	IconButton
+} from '@wordpress/components';
+
+import {
+	registerBlockType,
+	createBlock
+} from '@wordpress/blocks';
+
+import {
+	InnerBlocks,
+	InspectorControls
+} from '@wordpress/editor';
+
+import {
+	Component,
+	Fragment
+} from '@wordpress/element';
+
+import {
+	__
+} from '@wordpress/i18n';
+
+
+
+/**
+ * Components:
+ */
+
+class GrunionForm extends Component {
+	render() {
+		return (
+			<Fragment>
+				<InspectorControls>
+					<PanelBody title={ __( 'Submission Details', 'jetpack' ) }>
+						<TextControl
+							label={ __( 'What would you like the subject of the email to be?' ) }
+							value={ this.props.subject }
+							onChange={ ( x )=>this.props.setAttributes( { subject: x } ) }
+						/>
+						<TextControl
+							label={ __( 'Which email address should we send the submissions to?' ) }
+							value={ this.props.to }
+							onChange={ ( x )=>this.props.setAttributes( { to: x } ) }
+						/>
+						<TextControl
+							label={ __( 'What should the label on the form’s submit button say?' ) }
+							value={ this.props.submit_button_text ? this.props.submit_button_text : __( 'Submit »' ) }
+							onChange={ ( x )=>this.props.setAttributes( { submit_button_text: x } ) }
+						/>
+					</PanelBody>
+				</InspectorControls>
+				<div className="grunion-form">
+					{this.props.children}
+					<TextControl
+						className="button button-primary button-default grunion-submit-button"
+						value={ this.props.submit_button_text ? this.props.submit_button_text : __( 'Submit »' ) }
+						onChange={ ( x )=>this.props.setAttributes( { submit_button_text: x } ) }
```

Comment:
I've been getting some precommit warnings about using this method for setting the onchange -- do we have any recommendations for a better way to do this?

Reply:
I'm guessing it's because of the anonymous function, right? It's mostly for performance reasons due to unnecessary re-rendering. Better to move it to a function defined in the class. Here's a decent explanation: http://johnnyji.me/react/2016/06/27/why-arrow-functions-murder-react-performance.html